File: Android_lab2/app/src/main/java/com/example/android_lab2/MainActivity.java
package com.example.android_lab2;

import androidx.appcompat.app.AppCompatActivity;

import android.app.Activity;
import android.content.Intent;
import android.view.View;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button button = (Button) findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent newIntent = new Intent(MainActivity.this, SecondActivity.class);
                newIntent.putExtra("message","Введите ответ");
                startActivityForResult(newIntent, 10);
            }

        });

    }
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        Intent intent = getIntent();


        String firstNumb = textView.getText().toString();
        String secondNumb = editText.getText().toString();
        if (intent.getStringExtra("val") != null) {
            textView.setText(intent.getStringExtra("val"));
        }
        if (requestCode == 10) {
            if (requestCode == Activity.RESULT_OK) {
                EditText editText = (EditText) findViewById(R.id.editTextNumberSigned);
                TextView textView = (TextView) findViewById(R.id.textView);
                String result = data.getStringExtra("result");
                Integer Value = MyService.Sum(firstNumb, secondNumb);
                textView.setText(Value);
            }
        }
    }
}

File: lab1/app/src/main/java/by/bsu/lab_1/MainActivity.kt
package by.bsu.lab_1

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast


const val EXTRA_NUMBER = "by.bsu.number"

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var receivedMassage = intent.getStringExtra(EXTRA_NUMBER)
        val textView = findViewById<TextView>(R.id.textNumbView).apply {
            if(receivedMassage == null){
                receivedMassage = "0"
            }
            text = receivedMassage
        }
        var score = receivedMassage.toInt()

        val editTextNumberView = findViewById<EditText>(R.id.editTextNumb)

        val buttonOK = findViewById<Button>(R.id.buttonOk)
        buttonOK.setOnClickListener {
            val massageNumber = editTextNumberView.text.toString()

            if(massageNumber == "")
            {
                Toast.makeText(applicationContext, "Please, enter the number", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }
            else
            {
                score += massageNumber.toInt()
                val intent = Intent(this, SecondActivity::class.java).apply {
                    putExtra(EXTRA_NUMBER, score.toString())
                }
                startActivity(intent)
            }
        }
    }
}
